src/Symfony/Component/Routing/Tests/Generator/Dumper/CompiledUrlGeneratorDumperTest.php

```diff
@@ -231,4 +231,29 @@ public function testDumpWithSchemeRequirement()
         $this->assertEquals('https://localhost/app.php/testing', $absoluteUrl);
         $this->assertEquals('/app.php/testing', $relativeUrl);
     }
+
+    public function testDumpWithLocalizedRoutesPreserveTheGoodLocaleInTheUrl()
+    {
+        $this->routeCollection->add('foo.en', (new Route('/{_locale}/foo'))->setDefault('_locale', 'en')->setDefault('_canonical_route', 'foo'));
+        $this->routeCollection->add('foo.fr', (new Route('/{_locale}/foo'))->setDefault('_locale', 'fr')->setDefault('_canonical_route', 'foo'));
+        $this->routeCollection->add('fun.en', (new Route('/fun'))->setDefault('_locale', 'en')->setDefault('_canonical_route', 'fun'));
+        $this->routeCollection->add('fun.fr', (new Route('/amusant'))->setDefault('_locale', 'fr')->setDefault('_canonical_route', 'fun'));
+
+        file_put_contents($this->testTmpFilepath, $this->generatorDumper->dump());
+
+        $requestContext = new RequestContext();
+        $requestContext->setParameter('_locale', 'fr');
+
+        $compiledUrlGenerator = new CompiledUrlGenerator(require $this->testTmpFilepath, $requestContext, null, null);
+
+        $this->assertSame('/fr/foo', $compiledUrlGenerator->generate('foo'));
+        $this->assertSame('/en/foo', $compiledUrlGenerator->generate('foo.en'));
```

Comment:
I think I made a mistake and it should be `/fr/foo` here 😕

Reply:
Asking for the en route but request context locale is fr. Didn't specify a _locale parameter. Should fall back on fr.

Reply:
nope: you explicitly asked for the `en` route, so `en` it should be